customerio/customerio-ios
# docs: [vision-on-sample-app][9/n] SDK Installation view

docs: [vision-on-sample-app][9/n] SDK Installation view

## Context

This is a PR in a series of PRs to build sample app that runs Identify and Track in VisionPro
The app is built to be an interactive walk through guidance for developers who are integrating Customer.io Swift SDK in their VisioPro apps.
For more context:

- See the project [one pager](https://www.notion.so/custio/Spacial-Identify-and-Tracking-6587d848bbde494095c0585237326ed3?pvs=4)
- See [Linear Issue](https://linear.app/customerio/issue/MBL-115/sample-app-for-visionpro)

## PR Summary
This PR introduces a very brief SDK installation tutorial screen. It is intentially very brief as assume the developer knows how to use Swift Package Manager or Cocoapods if their app uses any of them already.

Beside the regular code review, I would really appreciate any suggestion to improve the visual content.


## Test plan

Beside testing the newly added screenand it associated behavior, since this is the first change in the app where we have multiple screen, I will also add a test to ensure the app remembers the last opened screen.

### New user experience
This scenario validates that:
1- Top left menu "opened" status is updated when we visit the SDK Installation screen correctly.
2- Taping/chosing the "installing CustomerIO SDK" direct to the correct screen
3- Tapping/chosing the "SDK Installtion" menu entry navigates to the correct screen

**Pre condition:**
Make sure the app is not installed on the device or simulator to run through the new user experience.

**Test steps**
- Open `VisionOs.xcproject`
- Hit CMD+R to run the project and make sure you have VisionOS simulator installed and selected
- Expected behavior:
   - The app opens on the App Intro View.
   - Top left menu shows no checkmark beside the "SDK Installation" entry because we never opened it before.
   - Tapping/chosing the "installing CustomerIO SDK" should open the SDK Installation View
   - Now the top left button should show the checkmark beside the "SDK Installation"


https://github.com/customerio/customerio-ios/assets/142905621/c4faca1a-7c4a-4550-8bb6-0dc478c948f9


### Existing user experience
This scenario validates that the app opens on the last visited screen and the menu state is persisted cross app runs.

https://github.com/customerio/customerio-ios/assets/142905621/ca9670d4-ee28-4277-8611-6ef31aff666b

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/customerio/customerio-ios/pull/597).
* #602
* #598
* __->__ #597
* #596
* #586
* #585
* #581
* #580
* #579
* #578
